server/src/main/java/io/crate/expression/reference/doc/lucene/BooleanColumnReference.java

```diff
@@ -29,38 +29,41 @@
 import org.elasticsearch.index.fielddata.SortedBinaryDocValues;
 
 import java.io.IOException;
+import java.io.UncheckedIOException;
 
 public class BooleanColumnReference extends LuceneCollectorExpression<Boolean> {
 
     private static final BytesRef TRUE_BYTESREF = new BytesRef("1");
     private final String columnName;
     private SortedBinaryDocValues values;
-    private Boolean value;
+    private int docId;
 
     public BooleanColumnReference(String columnName) {
         this.columnName = columnName;
     }
 
     @Override
     public Boolean value() {
-        return value;
+        try {
+            if (values.advanceExact(docId)) {
+                switch (values.docValueCount()) {
+                    case 1:
+                        return values.nextValue().compareTo(TRUE_BYTESREF) == 0;
+
+                    default:
+                        throw new GroupByOnArrayUnsupportedException(columnName);
+                }
+            } else {
+                return null;
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
     }
 
     @Override
     public void setNextDocId(int docId) throws IOException {
```

Comment:
Looks like we could remove the exception annotation from the parent class?

Reply:
yep you're right. I actually had also missed the short column reference. Pushed a fixup